frontend/model/notifications/selectors.js

```diff
@@ -39,17 +39,41 @@ sbp('sbp/selectors/register', {
       timestamp: data.createdDate ? new Date(data.createdDate).getTime() : Date.now(),
       type
     }
+    const rootState = sbp('chelonia/rootState')
+    if (!rootState.notifications) rootState.notifications = { items: [], status: {} }
+    if (rootState.notifications.items.some(item => item.hash === notification.hash)) {
+      // We cannot throw here, as this code might be called from within a contract side effect.
+      return console.error('This notification is already in the store.')
+    }
+    const index = rootState.notifications.items.findLastIndex(item => item < notification.timestamp)
+    rootState.notifications.items.splice(Math.max(0, index), 0, notification)
     sbp('gi.actions/identity/kv/addNotificationStatus', notification)
-    sbp('state/vuex/commit', keys.ADD_NOTIFICATION, notification)
+    sbp('okTurtles.events/emit', NOTIFICATION_EMITTED, notification)
   },
   'gi.notifications/markAsRead' (notification: Notification) {
     sbp('gi.actions/identity/kv/markNotificationStatusRead', notification.hash)
   },
   'gi.notifications/markAllAsRead' (groupID: string) {
-    const notifications = groupID
-      ? sbp('state/vuex/getters').unreadGroupNotificationsFor(groupID)
-      : sbp('state/vuex/getters').currentUnreadNotifications
-    const hashes = notifications.map(item => item.hash)
+    const rootState = sbp('chelonia/rootState')
+    if (!rootState.notifications) return
+    const hashes = rootState.notifications.items.filter(item => {
+      return !item.read && (!groupID || !item.groupID || item.groupID === groupID)
+    }).map(item => item.hash)
     sbp('gi.actions/identity/kv/markNotificationStatusRead', hashes)
+  },
+  'gi.notifications/remove' (hashes: string | string[]) {
+    if (!Array.isArray(hashes)) hashes = [hashes]
+    const rootState = sbp('chelonia/rootState')
+    if (rootState.notifications) return
```

Comment:
This seems like a bug?

Reply:
Do you mean that it should be `if (!rootState.notifications) return`? If so, you're right. Good catch.

If you mean that `rootState.notifications` would always be defined, that's not true. Calling `chelonia/reset` or just starting up the app would clear it.

If you mean something else, please explain (you did find a bug regardless).

Reply:
Yes, I meant the condition should be negated